lib/Importer.js

```diff
@@ -329,62 +310,55 @@ export default class Importer {
     return undefined;
   }
 
-  /**
-   * @param {Array} jsModules
-   * @param {String} variableName
-   * @return {JsModule}
-   */
-  resolveModuleUsingCurrentImports(jsModules, variableName) {
+  resolveModuleUsingCurrentImports(
+    jsModules: Array<JsModule>,
+    variableName: string
+  ): ?JsModule {
     if (jsModules.length === 1) {
       return jsModules[0];
     }
 
     // Look at the current imports and grab what is already imported for the
     // variable.
     let matchingImportStatement;
-    this.findCurrentImports().imports.forEach((ist) => {
+    this.findCurrentImports().imports.forEach((ist: ImportStatement) => {
       if (variableName === ist.defaultImport ||
          (ist.namedImports && ist.namedImports.indexOf(variableName) !== -1)) {
         matchingImportStatement = ist;
       }
     });
 
-    if (matchingImportStatement) {
-      if (jsModules.length === 0) {
-        // We couldn't resolve any module for the variable. As a fallback, we
-        // can use the matching import statement. If that maps to a package
-        // dependency, we will still open the right file.
-        let hasNamedExports = false;
-        if (matchingImportStatement.hasNamedImports()) {
-          hasNamedExports =
-            matchingImportStatement.namedImports.indexOf(variableName) !== -1;
-        }
-
-        const matchedModule = new JsModule({
-          importPath: matchingImportStatement.path,
-          hasNamedExports,
-        });
-
-        return matchedModule;
-      }
+    if (!matchingImportStatement) {
+      // Fall back to asking the user to resolve the ambiguity
+      return this.resolveOneJsModule(jsModules, variableName);
+    }
 
+    if (jsModules.length > 0) {
       // Look for a module matching what is already imported
-      return jsModules.find(jsModule => (
-        matchingImportStatement.path === jsModule.importPath
+      const { path } = matchingImportStatement;
+      return jsModules.find((jsModule: JsModule): boolean => (
+        path === jsModule.importPath
       ));
     }
 
-    // Fall back to asking the user to resolve the ambiguity
-    return this.resolveOneJsModule(jsModules, variableName);
+    // We couldn't resolve any module for the variable. As a fallback, we
+    // can use the matching import statement. If that maps to a package
+    // dependency, we will still open the right file.
+    let hasNamedExports = false;
+    if (matchingImportStatement.hasNamedImports()) {
+      hasNamedExports =
+        matchingImportStatement.namedImports.indexOf(variableName) !== -1;
+    }
+
+    const matchedModule = new JsModule({
+      importPath: matchingImportStatement.path,
+      hasNamedExports,
+    });
+
+    return matchedModule;
   }
```

Comment:
Thanks for making this method better. It was a little hard to follow along with the diff though.